dali/pipeline/operator/eager_operator.h

```diff
@@ -165,9 +168,18 @@ class DLL_PUBLIC EagerOperator {
                        ", instance name: \"", name_, "\", encountered:\n", what);
   }
 
-  static inline int GetDefaultNumThreads() {
-    int num_cores = std::thread::hardware_concurrency();
-    return num_cores < 6 ? num_cores : 6;
+  static inline std::shared_ptr<ThreadPool> GetSharedThreadPool() {
+    std::lock_guard<std::mutex> lock(shared_thread_pool_mutex_);
```

Comment:
Nitpick: to avoid locking every time you call the method you could consider using double-checked locking.